src/tail/session.rs

```diff
@@ -62,13 +62,12 @@ impl Session {
                 let mut delay = delay_for(duration);
 
                 loop {
-                    if delay.is_elapsed() {
-                        let heartbeat_result = send_heartbeat(&target, &client, &tail_id).await;
-                        if heartbeat_result.is_err() {
-                            return heartbeat_result;
-                        }
-                        delay = delay_for(duration);
+                    delay.await;
```

Comment:
the difference here is simply the  number of times  we enter this loop; before it was  just constant and we kept running the  `.is_elapsed()` check; now we  enter the loop once per  delay/heartbeat.